tests/matchers/google.go

```diff
@@ -0,0 +1,67 @@
+package matchers
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/googleapis/gax-go/v2/apierror"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/format"
+	"github.com/onsi/gomega/types"
+	"google.golang.org/api/googleapi"
+)
+
+type beGoogleAPIErrorWithStatusMatcher struct {
+	expected int
+}
+
+func BeGoogleAPIErrorWithStatus(expected int) types.GomegaMatcher {
```

Comment:
Neat!

Reply:
This is actually used in `dns-operator-gcp`, however (infuriatingly) this GCP client wraps the error, and so this version of the matcher is _slightly_ different. I was thinking we can make it work for both cases and have a repo somewhere with custom matchers.